### ISSUE TITLE
create TRichText

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#322]
From https://github.com/phetsims/phet-io/issues/1046, I think it would be good to do this before too long, since I counted 53 usages of RichText in phet-io instrumented sims. We should get this into sims before we deliver these to clients. Currently the are treated as TNode's, which means their text is not editable like Text or MultilineText. 

Tagging @samreid for comment.

**Comment by samreid** [comment]
Sounds good to instrument soon and include with next PhET-iO sim.

**Comment by zepumph** [comment]
I created it, and made it tandemOptional, just like Text.js. I think this is reasonable, but there are many RichText's in Phet-io sims that are not currently instrumented, that we probably want to be. My hope is go back and instrument a lot of these, if not all of them. I'm still not sure if we want this to be tandemRequired at some point.

Also I noticed that RichText (and MultilineText) does not trigger a text changed event in the setText message. This means that currently the data stream looks differently for these different types of texts.

We should potentially solidify a 'text api' across all of our implementations of text (Text, HTMLText, RichText, MultilineText, LinkText, ShadowText, ...). Many of these are children of Text, but not all of them. Maybe it would be good to have a parent of all texts that could solidify this api. I will bring this up in a new issue though.

Right now know that there is no text specific data stream for Rich/Multiline texts.
